src/api/hooks/useAmountInput.tsx

```diff
@@ -1,16 +1,40 @@
 import React, {useState, useEffect} from "react";
 import AmountTextField from "../../components/AmountTextField";
+import {useGetAccountResources} from "./useGetAccountResources";
 
-// TODO - this will likely be very dependent per network
-// pull from on chain config 0x1::stake::ValidatorSetConfiguration
-function isValidAmount(amount: string): boolean {
+interface StakingConfigData {
+  minimum_stake: string;
+  maximum_stake: string;
+}
+
+function isValidAmount(
+  amount: string,
+  minimumAmount: number,
+  maximumAmount: number,
+): boolean {
   const amountNum = parseInt(amount);
-  return amountNum > 0;
+  return amountNum <= maximumAmount && amountNum >= minimumAmount;
 }
 
 const useAmountInput = () => {
```

Comment:
With the staking amount range logic added, this is now more of a `useStakingAmountInput`. Although this hook meant to be the general amount input hook to serve all kinds of amount. I don't have strong opinion on wether we should keep it that way though. It's okay if we want to make it a staking amount hook, maybe just change the name to reflect the current logic.

Reply:
Agree. let me work on it

Reply:
I think it makes more sense to have it as a `stake amount input` instead of a global one for now, I started moving the validation logic into a staking related component and ended up with a plain TextField material ui component, also because staking page is the only place we use the amount input I think we can have it as a staking related file for now